experimental/ast/nodes.go

```diff
@@ -109,7 +109,8 @@ func (n *Nodes) NewDeclDef(args DeclDefArgs) DeclDef {
 	if !args.Type.Nil() {
 		raw.ty = args.Type.raw
 	} else {
-		raw.ty = rawType(rawPath{args.Keyword.ID(), args.Keyword.ID()})
+		kw := rawPath{args.Keyword.ID(), args.Keyword.ID()}.With(n.Context)
+		raw.ty = wrapPath[TypeKind](kw)
```

Comment:
Maybe this should be `wrapRawPath`, so you don't have attach a context that gets ignored? In the other two call sites, can then pass `t.Path.raw`.